### ISSUE TITLE
BUG: Changing .neptune location is broken with NeptuneLogger

### ISSUE THREAD

**Issue by wouterzwerink** [neptune-ai/neptune-client#1379]
### Describe the bug
I have code that changes the location of .neptune according to the docs:
[A clear and concise description of what the bug is.
](https://docs.neptune.ai/api/changing_location_of_neptune_folder/)

After updating from an older neptune-client to the latest neptune (1.3.1), this broke completely.
Now, it crashes when trying to log something.

### Reproduction
Create run using the setup shown above.
Create lightning NeptuneLogger with the run as argument.
Try to log something.
Will add minimal code example later.

### Expected behavior
.neptune folder is moved, logging still works

### Traceback
```
    logger = pl.loggers.NeptuneLogger(
  File "/home/micromamba/lib/python3.10/site-packages/pytorch_lightning/loggers/neptune.py", line 258, in __init__
    self._retrieve_run_data()
  File "/home/micromamba/lib/python3.10/site-packages/pytorch_lightning/loggers/neptune.py", line 273, in _retrieve_run_data
    root_obj.wait()
  File "/home/micromamba/lib/python3.10/site-packages/neptune/metadata_containers/metadata_container.py", line 433, in wait
    self._op_processor.wait()
  File "/home/micromamba/lib/python3.10/site-packages/neptune/internal/operation_processors/async_operation_processor.py", line 122, in wait
    raise NeptuneSynchronizationAlreadyStoppedException()
neptune.exceptions.NeptuneSynchronizationAlreadyStoppedException:
```
### Environment
**The output of `pip list`:**
```
<output goes here>
```
**The operating system you're using: Amazon linux**
**The output of `python --version`: 3.10.6**

**Comment by wouterzwerink** [comment]
Yes! Interestingly, it seems to not happen when my git directory is clean, only when there are uncommitted changes.

**Comment by Blaizzy** [comment]
Thanks for the update. We identified the cause of the issue as related to a feature we introduced recently that allows you to track uncommitted changes.

**Comment by SiddhantSadangi** [comment]
Hello @wouterzwerink 👋 

We just released [neptune v1.4.0](https://github.com/neptune-ai/neptune-client/releases/tag/1.4.0) that fixes this 🎉 

Please let me know if this works for you 🙏

**Comment by wouterzwerink** [comment]
@SiddhantSadangi sounds great! Is the dataloader fork bug also fixed in 1.4? I don't see it in the changelog, but that bug is very impactful

**Comment by SiddhantSadangi** [comment]
Not yet, unfortunately 😞 
We are still discussing the best course of action there.
